### PR TITLE
make tab previews based on mouse idle time

### LINKED ISSUE #8860: Make tab preview based on mouse idle time
## Test plan
See https://github.com/brave/browser-laptop/pull/9887

After #12823 please add the NO DELAY (current name) as a check too for undelayed tab previews. Thanks!!

<hr>

## Original issue description
Sometimes tab preview happen unintentionally if you go from urlbar to webview. Tab previews are based on RegExp that checks which element has hovered before current tab preview. 

After taking #3271 this still seems to be the best approach currently but we need to cover other elements that were missing on initial implementation.

### PR DESCRIPTION
**Pull request by cezaraugusto**
# This PR changed its purpose, please see description

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Close #8860

## Note for reviewers:

Previews are now happening based on mouse idle time. Which means that it will only be fired if you stay the mouse over a tab for a given amount of time -- which is defined in prefs->tabs.

Once the mouse is long enough over a tab and you wish to preview another tab, the preview is fired immediately. Once you leave the tab to another component, the event is canceled and you'll only be able to preview again if the mouse is idle again.

Test Plan:
### Intentional preview
1. Open three tabs
2. Hover over one inactive tab, let your mouse stopped for 2s (this is the default for now)
3. Tab should preview
4. Hover over third tab
5. Third tab should preview

### Unintentional preview/intentional preview
1. Open three tabs
2. Keep hovering from top/bottom, left/right
3. No tab should preview
4. Stop at a given tab
5. It should preview
6. Hover over something other than another tab
7. Hover over a tab again
8. It should not preview
9. Let your mouse stopped over a tab
10. It should preview

### Dealing with options
1. Go to prefs->tabs
2. Disable tab preview
3. Tab preview timing selection should be hidden
4. Enable again
5. Switch to "normal" (set to 1s) or "short" (set to 500ms)
6. Open three tabs
7. Hover over one inactive tab, let your mouse stopped for 2s (this is the default for now)
8. Tab should preview
9. Hover over third tab
10. Third tab should preview faster than previous STR

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run indivually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)

### CODE REVIEW

**Comment by cezaraugusto** [comment]
setting to work in progress since we'll fully change how we deal with tab previews

**Comment by cezaraugusto** [comment]
PR is still WIP as it needs Brad approval regarding preview timing but is ready for code review and feedback

**Comment by bsclifton** [comment]
@cezaraugusto can we get a rebase on this one please? 😄

**Comment by cezaraugusto** [comment]
ok rebased

File: js/actions/windowActions.js [hunk 1]
```diff
@@ -255,6 +255,19 @@ const windowActions = {
   },
 
   /**
+   * Dispatches a message to the store to set preview mode.
+   * This will check whether setPreviewFrame should be fired or not.
+   *
+   * @param {Boolean} shouldEnablePreview - true if user enters in previewMode state
+   */
+  setPreviewMode: function (shouldEnablePreview) {
```
Comment (bridiver) [inline js/actions/windowActions.js]: this action doesn't provide any new information that we don't already have in `setPreviewFrame`. We are already duplicating information in `setTabHoverState` and `setPreviewFrame` (fixed in #9542) and definitely don't want to add any more actions for this functionality

**Comment by cezaraugusto** [comment]
moving to 0.18.x per last weekly triage meeting. WIP and blocked until #9542 lands, which is set for the same milestone.

**Comment by cezaraugusto** [comment]
am going to close this and keep as a ref and open a new one with updated logic

**Pull request closed without merging**